client/base.py

```diff
@@ -45,10 +46,11 @@ def _post(self, path: str, data: Any) -> requests.Response:
         self.__check_response(response)
         return response
 
-    def _delete(self, path: str) -> requests.Response:
+    def _delete(self, path: str, reason: str) -> requests.Response:
```

Comment:
I believe that this is a generic delete method for all resources, not just _jobs_? If would probably make more sense to make this parameter into something like `json: Any = None` and pass in the structured object.